From 90dd1265542de0cb289e43e2ec6c2e0b7b79bf03 Mon Sep 17 00:00:00 2001
From: Lee Wells <lee_wells@jabil.com>
Date: Tue, 10 Oct 2023 15:13:40 -0400
Subject: [PATCH] yocto: Don't use hardwired paths to libusb-1.0

CMake finds libusb-1.0 in a sysroot directory which is private
to the xlink package.  This breaks builds of client packages
if the workdirs for xlink are not present (either using rm-work,
or getting an sstate cache hit for xlink)

Replace this found path with a simple -lusb-1.0.  This will let
the linker find the package in the correct yocto paths for client
packages.  This is how the zeromq recipe in meta-oe fixes a similar
problem with libsodium.
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2160770..600f2f3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,7 +69,7 @@ if(XLINK_LIBUSB_SYSTEM)
         PRIVATE "${LIBUSB_INCLUDE_DIR}"
     )
     target_link_libraries(${TARGET_NAME}
-        PUBLIC ${LIBUSB_LIBRARY}
+        PUBLIC -lusb-1.0
     )
 else()
     # Link to CMake libusb
-- 
2.34.1

